docs/guide/02-Introducing-Components.md

```diff
@@ -0,0 +1,316 @@
+
+```purs
+initialState :: forall i. i -> State
+initialState _ = 0
+```
+
+### Actions
+
+Halogen components can update state, perform effects, and communicate with other components in response to events that arise internally. Components use an "action" type to describe what kinds of things a component can do in response to internal events.
+
+Our counter has two internal events: a click event on the button to decrement the count and a click event on the button to increment the count. We can describe what our component should do in response to these events using a data type we'll call `Action`:
+
+```purs
+data Action = Increment | Decrement
+```
+
+This type signifies that our component is capable of incrementing and decrementing. In a moment, we'll see this type used in our HTML -- another example of Halogen's declarative nature.
+
+Just like how our state type had to be paired with an `initialState` function that describes how to produce a `State` value, our `Action` type should be paired with a function called `handleAction` that describes what to do when one of these actions occurs.
+
+```purs
+handleAction :: forall o m. Action -> H.HalogenM State Action () o m Unit
+```
+
+As with our input type, we can leave type variables open for types that we aren't using. 
+
+* The type `()` means our component has no child components. We could also leave it open as a type variable, because we aren't using it, but `()` is so short that you'll see this commonly used.
+* The `o` type parameter is only used when your component communicates with a parent.
+* The `m` type parameter is only relevant when your component performs effects.
+
+Since our counter has no child components we'll use `()` to describe them, and because it doesn't communicate with a parent or perform effects we'll leave the `o` and `m` type variables open.
+
+Here's the `handleAction` function for our counter:
+
+```purs
+handleAction :: forall o m. Action -> H.HalogenM State Action () o m Unit
+handleAction = case _ of
+  Decrement -> 
+    H.modify_ \state -> state - 1
+
+  Increment ->
+    H.modify_ \state -> state + 1
+```
+
+Our `handleAction` function responds to `Decrement` by reducing our state variable by 1, and to `Increment` by increasing our state variable by 1. Halogen provides several update functions you can use in your `handleAction` function; these ones are commonly used:
+
+* `modify` allows you to update the state, given the previous state, returning the new state
+* `modify_` is the same as `modify`, but it doesn't return the new state (thus you don't have to explicitly discard the result, as you would with `modify`)
+* `get` allows you to retrieve the current state
+* `gets` allows you to retrieve the current state and also apply a function to it (most commonly, `_.property` to retrieve a particular field from a record)
```

Comment:
> most commonly, `_.property` to 

Should this be "`_.fieldName` to retrieve a particular field from a record" ? Or rephrase it to, "`_.property` to retrieve a field name `property` from a record."

Reply:
```suggestion
* `gets` allows you to retrieve the current state and also apply a function to it (most commonly, `_.fieldName` to retrieve a particular field from a record)
```